pkg/kclient/errors.go

```diff
@@ -19,3 +19,12 @@ type DeploymentNotFoundError struct {
 func (e *DeploymentNotFoundError) Error() string {
 	return fmt.Sprintf("deployment not found for the selector: %s", e.Selector)
 }
+
+// ServiceNotFoundError returns an error if no service is found with the selector
+type ServiceNotFoundError struct {
+	Selector string
+}
+
+func (e *ServiceNotFoundError) Error() string {
+	return fmt.Sprintf("service not found for the selector: %s", e.Selector)
```

Comment:
Could you use `service not found for the selector %q` instead? (to reserve `...: ...` for wrapped error)